ui/apps/platform/src/Containers/Audit/ListeningEndpoints/hooks/useDeploymentListeningEndpoints.tsx

```diff
@@ -0,0 +1,34 @@
+import { useCallback } from 'react';
+import { UsePaginatedQueryReturn, usePaginatedQuery } from 'hooks/usePaginatedQuery';
+import { listDeployments } from 'services/DeploymentsService';
+import {
+    ProcessListeningOnPort,
+    getListeningEndpointsForDeployment,
+} from 'services/ProcessListeningOnPortsService';
+import { ListDeployment } from 'types/deployment.proto';
+
+const sortOptions = { field: 'Deployment', reversed: 'false' };
+const pageSize = 10;
+
+/**
+ * Returns a paginated list of deployments with their listening endpoints.
+ */
+export function useDeploymentListeningEndpoints(): UsePaginatedQueryReturn<
+    ListDeployment & { listeningEndpoints: ProcessListeningOnPort[] }
+> {
+    const queryFn = useCallback((page: number) => {
+        return listDeployments({}, sortOptions, page, pageSize).then((res) => {
+            return Promise.all(
+                res.map((deployment) => {
+                    const { request } = getListeningEndpointsForDeployment(deployment.id);
```

Comment:
Yes agreed, unfortunately this is only one of two enhancements in 4.2 that are going to have the n+1 query problem on the front end. There are plans to integrate the data (listening endpoints & baselines) so that we don't need to query in this way, but I haven't heard specifics on what the API changes will be. I'll bring it up during one of our syncs to see if the BE team has any concrete plans yet.

AFAIK though, nothing is planned to be done for the 4.2 release by the BE teams.